### PR TITLE
Feature/LINGUINI-106: create aside menu, reusable card for each topic, topics list in the midle-right side of the page, search subtopics functionality, featured articles section which will contain Medium greatest articles regarding the chosen main topic

### PR DESCRIPTION
**Pull request by BogdanBlanaru**
create the aside menu and a service which will handle the topics

### CODE REVIEW

File: src/styles.scss [hunk 1]
```diff
@@ -5,3 +5,16 @@
 @tailwind utilities;
 
 // add here the custom classes which you don t find with tailwind
```
Comment (radumaftei) [inline src/styles.scss:10]: Which you dont founded *****!!!*!**!**!!!!!!!!!!!

File: src/app/services/topic.service.ts [hunk 2]
```diff
@@ -1,9 +1,232 @@
...
 
-  constructor() { }
+  constructor() {}
+
+  getListOfTopics(): Topic[] {
+    return this.TOPICSLIST;
+  }
+
+  getListOfSubtopicsByTopicName(topic: string): SubTopic[] | undefined {
+    return this.TOPICSLIST.filter((el) => el.name === topic)[0]?.subtopic;
```
Comment (radumaftei) [inline src/app/services/topic.service.ts]: I dont really like this approach. If filter is empty you can return empty array. Undefined means that we dont have elements in the filtered array, but that means that we have an empty array. We will have less checks in the template if we use only arrays.

File: src/app/services/topic.service.ts [hunk 3]
```diff
@@ -1,9 +1,232 @@
...
+  }
+
+  getSubtopicBySearchedValue(
+    searchedValue: string,
+    topicName: string
+  ): SubTopic[] | undefined {
+    const listOfSubtopics = this.getListOfSubtopicsByTopicName(topicName);
+    return listOfSubtopics?.filter(
```
Comment (radumaftei) [inline src/app/services/topic.service.ts]: Same here

File: src/app/services/topic.service.ts [hunk 4]
```diff
@@ -1,9 +1,232 @@
...
+      (el) =>
+        el.title.toLowerCase().includes(searchedValue) ||
```
Comment (radumaftei) [inline src/app/services/topic.service.ts:231]: searchedValue should also be toLowerCase(). Do it only once and reuse that value from a variable

File: src/app/components/subtopics-list/subtopics-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  selector: 'app-subtopics-list',
+  standalone: true,
+  imports: [CommonModule, SubtopicComponent, SearchSectionComponent],
+  providers: [TopicService],
+  templateUrl: './subtopics-list.component.html',
+  styleUrl: './subtopics-list.component.scss',
+})
+export class SubtopicsListComponent implements OnChanges {
+  @Input() topicName: string = '';
+  protected subtopics: SubTopic[] | undefined;
```
Comment (radumaftei) [inline src/app/components/subtopics-list/subtopics-list.component.ts]: Dont forget to remove  | undefined in here after you refactor in topic.service.ts. Hint: If you ever wanna use undefined as a value in a class property, just do it like this: 
`protected subtopics?: SubTopic[]`

File: src/app/components/subtopics-list/subtopics-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  protected searchedSubtopic: string | undefined;
+
+  constructor(private topicService: TopicService) {}
+
+  ngOnChanges(): void {
+    this.subtopics = this.topicService.getListOfSubtopicsByTopicName(
```
Comment (radumaftei) [inline src/app/components/subtopics-list/subtopics-list.component.ts]: At the moment this is the only `@Input` but what if we add more `@Inputs` to the class. Please insert the changes: SimpleChanges in the params of the method, and only change the this.subtopics array if that input is received.

File: src/app/components/subtopics-list/subtopics-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      this.topicName
+    );
+  }
+
+  searchSubtopic(event: string) {
```
Comment (radumaftei) [inline src/app/components/subtopics-list/subtopics-list.component.ts:28]: Have a check for empty inputs. You dont wanna process anything if the user deleted everything from the input

File: src/app/components/subtopics-list/subtopics-list.component.html [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<app-search-section
+  *ngIf="topicName"
```
Comment (radumaftei) [inline src/app/components/subtopics-list/subtopics-list.component.html]: Dare to use the new directives for *ngIf and *ngFor
https://angular.dev/api/core/@if#mat-tab-content-88-0
https://angular.dev/api/core/@for#mat-tab-content-88-0

File: src/app/components/subtopics-list/subtopic/subtopic.component.ts [hunk 9]
```diff
@@ -1,13 +1,14 @@
...
   selector: 'app-subtopic',
   standalone: true,
   imports: [CommonModule],
   templateUrl: './subtopic.component.html',
-  styleUrl: './subtopic.component.scss'
+  styleUrl: './subtopic.component.scss',
 })
 export class SubtopicComponent {
-
+  @Input() subtopicName: SubTopic | undefined;
```
Comment (radumaftei) [inline src/app/components/subtopics-list/subtopic/subtopic.component.ts]: hmmhmm

File: src/app/components/subtopics-list/search-section/search-section.component.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './search-section.component.html',
+  styleUrl: './search-section.component.scss',
+})
+export class SearchSectionComponent implements OnInit {
+  protected searchedTitle: string = '';
+  @Output() searchedValueSent = new EventEmitter<string>();
+
+  @ViewChild('subtopicsSearchValue', { static: true })
```
Comment (radumaftei) [inline src/app/components/subtopics-list/search-section/search-section.component.ts]: static:true is by default, hence you can remove it

File: src/app/components/subtopics-list/search-section/search-section.component.ts [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  subtopicsSearchValue!: ElementRef;
+
+  ngOnInit(): void {
+    fromEvent(
```
Comment (radumaftei) [inline src/app/components/subtopics-list/search-section/search-section.component.ts]: fromEvent represents a hot Observable, which means the value is produced outside of its control ( you either subscribe at that moment, or you lose that value ). Use a subscription initialized on top of the class, `private readonly subscription = new Subscription()` and unsubscribe from the obs.

**Review by radumaftei — CHANGES_REQUESTED** [review]

File: src/app/components/subtopics-list/search-section/search-section.component.ts [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+  imports: [CommonModule],
+  templateUrl: './search-section.component.html',
+  styleUrl: './search-section.component.scss',
+})
+export class SearchSectionComponent implements OnInit {
+  protected searchedTitle: string = '';
+  @Output() searchedValueSent = new EventEmitter<string>();
+
+  @ViewChild('subtopicsSearchValue', { static: true })
+  subtopicsSearchValue!: ElementRef;
```
Comment (radumaftei) [inline src/app/components/subtopics-list/search-section/search-section.component.ts]: Dont use Non-null Assertion Operator, what if the view is not initialized?

**Review by radumaftei — CHANGES_REQUESTED** [review]

File: src/app/services/topic.service.ts (hunk 4 above)
Comment (BogdanBlanaru) [reply src/app/services/topic.service.ts:231]: I did it here: It s ok?
image

File: src/app/components/subtopics-list/subtopics-list.component.html (hunk 8 above)
Comment (BogdanBlanaru) [reply src/app/components/subtopics-list/subtopics-list.component.html]: I tried but for some reason, I don't know it doesn't work. You can also see here that I commented on the code with the attempt.
image

File: src/app/components/subtopics-list/search-section/search-section.component.ts (hunk 10 above)
Comment (BogdanBlanaru) [reply src/app/components/subtopics-list/search-section/search-section.component.ts:29]: Yes. I saw that from angular 8 it is but it's not working without static: true. 
image

File: src/app/services/topic.service.ts (hunk 4 above)
Comment (radumaftei) [reply src/app/services/topic.service.ts:231]: Call ? Yes?

**Review by radumaftei — APPROVED** [review]

**Pull request merged**